kythe/cxx/indexer/cxx/IndexerASTHooks.cc

```diff
@@ -2671,6 +2671,17 @@ bool IndexerASTVisitor::VisitVarDecl(const clang::VarDecl* Decl) {
   if (auto TyNodeId = BuildNodeIdForType(Decl->getType())) {
     Observer.recordTypeEdge(DeclNode, *TyNodeId);
   }
+  if (options_.RecordVariableInitTypes && Decl->hasInit()) {
+    if (const auto* init = Decl->getInit(); init != nullptr) {
+      auto ty = init->getType();
+      if (!ty.isNull()) {
+        if (auto init_ty = BuildNodeIdForType(ty)) {
+          Observer.recordInitTypeEdge(DeclNode, *init_ty);
+          Observer.recordFlatSource(*init_ty, ty.getAsString());
```

Comment:
Why the "flat source" separately rather than just pointing to the type node and using its marked source?